src/Model/lbsObjectCommunicator.py

```diff
@@ -47,14 +47,11 @@
 
 
 class ObjectCommunicator:
-    """A class holding received and sent messages for an object
+    """ A class holding received and sent messages for an object
     """
 
     def __init__(self, r: dict = None, s: dict = None, i=None, logger: Logger = None):
-        if r is None:
-            r = {}
-        if s is None:
-            s = {}
+        # TODO: if i is needed? What for? i is used only when populate from samplers
```

Comment:
Yes, this is needed for a sanity check in `summarize_unidirectional()`:
```
            if k.get_id() == self.object_index:
                self.lgr.error(f"object {self.object_index} cannot send communication to itself.")
                sys.exit(1)
```
What is however a problem is that is **not** used when we are populating from the VT reader. This is probably what should be addressed.